pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/InvertedIndexAndDictionaryBasedForwardIndexCreator.java

```diff
@@ -280,11 +282,16 @@ private Map<String, String> createForwardIndexForSVColumn()
       writeToForwardIndex(dictionary, context);
 
       // Setup and return the metadata properties to update
-      Map<String, String> metadataProperties = new HashMap<>();
-      metadataProperties.put(getKeyFor(_columnName, HAS_DICTIONARY), String.valueOf(_dictionaryEnabled));
-      metadataProperties.put(getKeyFor(_columnName, DICTIONARY_ELEMENT_SIZE),
-          String.valueOf(_dictionaryEnabled ? _columnMetadata.getColumnMaxLength() : 0));
-      return metadataProperties;
+      if (_dictionaryEnabled) {
+        return Map.of();
```

Comment:
why don't we need to set DICTIONARY_ELEMENT_SIZE here ? I guess HAS_DICTIONARY isn't set because its default value is true.

Reply:
We don't need to update any field because it already has dictionary